src/cli.rs

```diff
@@ -274,189 +292,239 @@ pub fn handle_cli(cli: &Cli) {
+            }
+            EdgeAppCommands::Publish { path } => {
+                let authentication = Authentication::new();
+                let command = commands::edge_app::EdgeAppCommand::new(authentication);
+                match command.publish(transform_edge_app_path_to_manifest(path).as_path()) {
+                    Ok(manifest) => {
+                        println!("Edge app successfully published. Your manifest: {manifest:?}");
                     }
                     Err(e) => {
-                        error!("Error occurred: {:?}", e);
-                        std::process::exit(1);
+                        println!("Failed to publish edge app manifest: {e}.");
                     }
                 }
             }
         },
-        Commands::Asset(command) => match command {
-            AssetCommands::List { json } => {
-                let asset_command = commands::asset::AssetCommand::new(authentication);
-                handle_command_execution_result(asset_command.list(), json);
-            }
-            AssetCommands::Get { uuid, json } => {
-                let asset_command = commands::asset::AssetCommand::new(authentication);
-                handle_command_execution_result(asset_command.get(uuid), json);
-            }
-            AssetCommands::Add { path, title, json } => {
-                let asset_command = commands::asset::AssetCommand::new(authentication);
-                handle_command_execution_result(asset_command.add(path, title), json);
-            }
-            AssetCommands::Delete { uuid } => {
-                let asset_command = commands::asset::AssetCommand::new(authentication);
-                match get_asset_title(uuid, &asset_command) {
-                    Ok(title) => {
-                        info!("You are about to delete the asset named \"{}\".  This operation cannot be reversed.", title);
-                        info!("Enter the asset title to confirm the asset deletion: ");
-                        io::stdout().flush().unwrap();
-
-                        let stdin = io::stdin();
-                        let mut user_input = String::new();
-                        match stdin.read_line(&mut user_input) {
-                            Ok(_) => {}
-                            Err(e) => {
-                                error!("Error occurred: {}", e);
-                                std::process::exit(1);
-                            }
-                        }
+    }
+}
 
-                        if title != user_input.trim() {
-                            error!("The title you entered is incorrect. Aborting.");
+fn transform_edge_app_path_to_manifest(path: &Option<String>) -> PathBuf {
+    let mut p = "screenly.yml".to_owned();
+    if let Some(path_input) = path {
+        let mut buf = PathBuf::from(path_input);
+        buf.push("screenly.yml");
+        p = buf.to_str().unwrap_or("screenly.yml").to_owned();
+    }
+
+    PathBuf::from(p)
+}
```

Comment:
I'd say that this would look more natural with combinators. 

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=8b2253fedf2f34b607711f4d81bc853b
```suggestion
fn transform_edge_app_path_to_manifest(path: &Option<String>) -> PathBuf {
    let mut result = PathBuf::from(
        path_input.as_ref().map(String::as_str).unwrap_or("")
    );
    result.push("screenly.yml");
    result
}
```